### PR TITLE
[Test] Added kubernetes 1.8 stage on Jenkins

### PR DESCRIPTION
**Pull request by eloycoto**
Hello! 

Added support to k8s 1.8 on Ginkgo Tests. 

Regards

### CODE REVIEW

**Comment by ianvernon** [comment]
Looks great!
One concern I have here is how this will affect the Jenkins slaves w.r.t. load. Currently we have two executors per node, and each node has 8 cores. Given that the Ginkgo build takes 30 minutes as opposed to 1 hour 45 minutes with the bash script based build, maybe we can go with one executor per node instead of two executors per node to allow for each build to use more resources. Alternatively, maybe we can test other K8s versions only in nightly builds of master, and test only the latest version of K8s that we support in merge builds. This would allow us to increase the number of executors per node, as we would be using fewer resources per build. @tgraf , @aanm: thoughts?

**Comment by tgraf** [comment]
> One concern I have here is how this will affect the Jenkins slaves w.r.t. load. Currently we have two executors per node, and each node has 8 cores. Given that the Ginkgo build takes 30 minutes as opposed to 1 hour 45 minutes with the bash script based build, maybe we can go with one executor per node instead of two executors per node to allow for each build to use more resources. Alternatively, maybe we can test other K8s versions only in nightly builds of master, and test only the latest version of K8s that we support in merge builds. This would allow us to increase the number of executors per node, as we would be using fewer resources per build. @tgraf , @aanm: thoughts?

Let's used dedicated nodes. It also leads to more predictable performance results.

**Comment by aanm** [comment]
@ianvernon Can't we test the k8s builds for only k8s changes?

File: ginkgo.Jenkinsfile [hunk 1]
```diff
@@ -52,6 +55,7 @@ pipeline {
                     junit 'test/*.xml'
                     sh 'cd test/; vagrant destroy -f'
                     sh 'cd test/; K8S_VERSION=1.6 vagrant destroy -f'
+                    sh 'cd test/; K8S_VERSION=1.8 vagrant destroy -f'
```
Comment (aanm) [inline ginkgo.Jenkinsfile:58]: Shouldn't be a line with `1.7`?

**Review by aanm — COMMENTED** [review]
We need to generate the daemonsets on the fly based on the `examples/kubernetes` because we might change the DS without realized the tests are using an old version or different set up and break compatibility with a K8S version.

**Comment by ianvernon** [comment]
> @ianvernon Can't we test the k8s builds for only k8s changes?

@aanm can you please elaborate? Are you saying that if there is only a change to the K8s tests, then do not run the other stages of the build?

**Comment by aanm** [comment]
>>@ianvernon Can't we test the k8s builds for only k8s changes?
>@aanm can you please elaborate? Are you saying that if there is only a change to the K8s tests, then do not run the other stages of the build?

No, you were asking if we could have nightly builds for k8s. I was asking if it wasn't possible to run the kubernetes runtime tests based on the labels of a PR, or the code changes.

File: ginkgo.Jenkinsfile (hunk 1 above)
Comment (eloycoto) [reply ginkgo.Jenkinsfile:58]: Hello. 

1.7 is the default env variable, so is deleted on the line 52

**Comment by eloycoto** [comment]
@aanm  Could the examples files be in Json format instead of yaml? 

In JSON format it's easier to replace, and from examples to manifest there are a few of changes. Etcd settings, geneve &vxlan so if it can be in JSON I can use dot notation to replace and render.

**Comment by aanm** [comment]
@eloycoto, can't parse the yamls to json with go? or even load yaml with go and replace the values as well?

**Comment by tgraf** [comment]
@eloycoto What's the state of this, can we review and merge it?

**Review by aalemayhu — COMMENTED** [review]
Please ping when this is ready for review.

**Comment by eloycoto** [comment]
I'll close this for now, let's get ginkgo fully tested and I'll re-open again ;-)

**Pull request closed without merging**